src/init/features/dataconnect/index.ts

```diff
@@ -18,20 +20,159 @@ const SCHEMA_TEMPLATE = readFileSync(join(TEMPLATE_ROOT, "schema.gql"), "utf8");
+      databaseId: info.cloudSqlDatabase,
+      enableGoogleMlIntegration: false,
+    });
+  }
+}
+
+function subValues(
+  template: string,
+  replacementValues: {
+    serviceId: string;
+    cloudSqlInstanceId: string;
+    cloudSqlDatabase: string;
+    connectorId: string;
+  },
+): string {
+  const replacements: Record<string, string> = {
+    serviceId: "__serviceId__",
+    cloudSqlDatabase: "__cloudSqlDatabase__",
+    cloudSqlInstanceId: "__cloudSqlInstanceId__",
+    connectorId: "__connectorId__",
+  };
+  let replaced = template;
+  for (const [k, v] of Object.entries(replacementValues)) {
+    replaced = replaced.replace(replacements[k], v);
+  }
+  return replaced;
+}
 
-  let cloudSqlInstanceId = "";
-  let newInstance = false;
-  let locationId = "";
+async function promptForService(setup: Setup, info: RequiredInfo): Promise<RequiredInfo> {
   if (setup.projectId) {
+    await ensureApis(setup.projectId);
+    // TODO: Support initing with services that have existing sources/files
+    const existingServices = await listAllServices(setup.projectId);
+    const existingServicesAndSchemas = await Promise.all(
+      existingServices.map(async (s) => {
+        return {
+          service: s,
+          schema: await getSchema(s.name),
+        };
+      }),
+    );
+    const existingFreshServicesAndSchemas = existingServicesAndSchemas.filter((s) => {
+      return !s.schema?.source.files?.length;
+    });
+    if (existingFreshServicesAndSchemas.length) {
+      const choices: { name: string; value: any }[] = existingFreshServicesAndSchemas.map((s) => {
+        const serviceName = parseServiceName(s.service.name);
+        return {
+          name: `${serviceName.location}/${serviceName.serviceId}`,
+          value: s,
+        };
+      });
+      choices.push({ name: "Create a new service", value: undefined });
+      const choice: { service: Service; schema: Schema } = await promptOnce({
+        message:
+          "Your project already has existing services. Which would you like to set up local files for?",
+        type: "list",
+        choices,
+      });
+      if (choice) {
+        const serviceName = parseServiceName(choice.service.name);
+        info.serviceId = serviceName.serviceId;
+        info.locationId = serviceName.location;
+        if (choice.schema) {
+          info.cloudSqlInstanceId =
+            choice.schema.primaryDatasource.postgresql?.cloudSql.instance ?? "";
+          info.cloudSqlDatabase = choice.schema.primaryDatasource.postgresql?.database ?? "";
+        }
+      }
+    }
+  }
+
+  if (info.serviceId === "") {
+    info.serviceId = await promptOnce({
+      message: "What ID would you like to use for this service?",
+      type: "input",
+      default: "my-service",
+    });
+  }
+  return info;
+}
+
+async function promptForCloudSQLInstance(setup: Setup, info: RequiredInfo): Promise<RequiredInfo> {
+  if (setup.projectId && info.cloudSqlInstanceId) {
```

Comment:
The caller already checked that `cloudSqlInstanceId` is missing. This branch isn't reachable.

```
  if (info.cloudSqlInstanceId === "") {
    info = await promptForCloudSQLInstance(setup, info);
  }
```

It looks right after Dropping this if-statement.

Reply:
Done.